### PR TITLE
29 - Home Page - Khadija

### PR DESCRIPTION
**Pull request by KhadijaHaji**
## Description

Write a summary of your change. Explain why you have made it.
Installed a Carousel Library -> npm install slick-carousel
Installes a react-scroll library -> npm install react-scroll
Created a Carousel component but I have yet to style it properly.
I have Created an events and news section, I have included both mobile and desktop version. In mobile version, for both sections the cards are in a column and then they are both in a row for mobile version.

## Issues
I need to spend some time making changes to the carousel.
I need to add media queries for the carousel and get involved section.

## Related to

issue #29 

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have carefully reviewed my own code
- [ ] I have commented my code
- [ ] I have updated any documentation

### CODE REVIEW

File: client/src/pages/HomePage/components/Carousel/Carousel.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	const settings = {
+		dots: true, // Display dots at the bottom indicating image count
+		arrows: true, // Display navigation arrows
+		autoplay: true, // Automatically transition to the next image
+		autoplaySpeed: 3000, // Delay between image transitions (in milliseconds)
+	};
+
+	return (
+		<Slider {...settings}>
+			<div className="slide-container">
```
Comment (ShayanMahnam) [inline client/src/pages/HomePage/components/Carousel/Carousel.jsx]: here we are iterating ourself. do you think we can create component for slide-container? and pass image src and alt as props? I'm not if map works inside this Slider component but its worth to try it

File: client/src/pages/HomePage/HomePage.jsx [hunk 2]
```diff
@@ -1,11 +1,23 @@
 import "./HomePage.css";
+import Carousel from "./components/Carousel/Carousel";
+import HomeCards from "./components/HomeCards/HomeCards";
+import GetInvolved from "./components/GetInvolved/GetInvolvedSection";
```
Comment (ShayanMahnam) [inline client/src/pages/HomePage/HomePage.jsx:5]: well done so far!

on the mobile screen's small size we have an issue I believe this SVG pushed everything and make other designs look bad 

Screenshot 2023-06-26 at 22 46 05

Comment (KhadijaHaji) [reply client/src/pages/HomePage/HomePage.jsx:5]: Thanks @ShayanMahnam we've now changed it into a span with icon. Issue has been resolved.

File: client/src/pages/HomePage/components/HomeGetInvolvedSection/HomeGetInvolvedSection.jsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import "./HomeGetInvolvedSection.css";
+import { Link } from "react-router-dom";
+import Line from "../../../../components/Line/Line";
+
+import { FaHandHoldingMedical, FaPerson } from "react-icons/fa6";
+
```
Comment (ShayanMahnam) [inline client/src/pages/HomePage/components/HomeGetInvolvedSection/HomeGetInvolvedSection.jsx]: we can use this for fundraise btn when we click its goes up to the other page
```suggestion
	const handleLinkClick = () => {
		window.scrollTo(0, 0);
	};
```

File: client/src/pages/HomePage/components/HomeGetInvolvedSection/HomeGetInvolvedSection.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const HomeGetInvolvedSection = () => {
+	return (
+		<section className="home-get-involved-section">
+			<h2>Get Involved</h2>
+			<Line />
+			<p>
+				Lorem Ipsum is simply dummy text of the printing and typesetting
+				industry. Lorem Ipsum has been the industrys standard...
+			</p>
+			<div className="home-get-involved-links-container">
```
Comment (ShayanMahnam) [inline client/src/pages/HomePage/components/HomeGetInvolvedSection/HomeGetInvolvedSection.jsx:23]: following last cmnt 
```suggestion
					onClick={handleLinkClick}
```

File: client/src/pages/HomePage/components/HomeCard/HomeCard.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+		content.length > 350 ? `${content.substring(0, 350)}...` : content;
+	return (
+		<section className="home-card-container">
+			<div className="home-card-image-container">
+				<img className="home-card-image" src={image} alt={altText} />
+			</div>
+			<div className="home-card-content">
+				<h2 className="home-card-content-title">{title}</h2>
+				<p className="home-card-content-text">{truncatedContent}</p>
+				<Link to="/get-involved" className="home-card-content-link">
```
Comment (ShayanMahnam) [inline client/src/pages/HomePage/components/HomeCard/HomeCard.jsx]: why crads link to get-involved page?

**Comment by KhadijaHaji** [comment]
@ShayanMahnam I made all the changes, can you check one more time and approve please :)

File: client/src/pages/HomePage/components/HomeHero/HomeHero.css [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+	padding: 10px 20px;
+}
+
+.slide-image-container {
+	height: 500px;
+	max-width: 1200px;
+}
+
+.slick-list {
+	height: 500px;
```
Comment (VitalinaKuzmenko) [inline client/src/pages/HomePage/components/HomeHero/HomeHero.css]:
```suggestion
	height: 350px;
```

File: client/src/pages/HomePage/components/HomeHero/HomeHero.css [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+	bottom: 50px;
+	width: 60%;
+	color: white;
+	font-size: 22px;
+	background-color: rgba(37, 36, 36, 0.517);
+	padding: 10px 20px;
+}
+
+.slide-image-container {
+	height: 500px;
```
Comment (VitalinaKuzmenko) [inline client/src/pages/HomePage/components/HomeHero/HomeHero.css]:
```suggestion
	height: 350px;
```

File: client/src/pages/HomePage/components/HomeHero/HomeHero.css [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+	color: white;
+}
+
+@media screen and (min-width: 768px) {
+	.slide-title {
+		left: 100px;
+		bottom: 70px;
+		width: 50%;
+		font-size: 30px;
+	}
```
Comment (VitalinaKuzmenko) [inline client/src/pages/HomePage/components/HomeHero/HomeHero.css:79]:
```suggestion
	}
	.slide-image-container {
		height: 500px;
	}

	.slick-list {
		height: 500px;
	}
```

**Comment by bazmurphy** [comment]
Note: 
We should try and get together to solve the issue with the cards (and how many) in different screen widths and how to handle that.
I think we could solve it together quickly.

But don't let that hold up this being merged.

**Review by ShayanMahnam — APPROVED** [review]

**Pull request merged**